mlir/tools/quantum-opt/quantum-opt.cpp

```diff
@@ -33,6 +33,10 @@
 #include "Quantum/IR/QuantumDialect.h"
 #include "Quantum/Transforms/Passes.h"
 
+namespace test {
+void registerTestDialect(mlir::DialectRegistry &);
+} // namespace test
+
```

Comment:
Minor, but is there no header file?

Reply:
Not one that is easy to link. I think it is done specifically because it is for testing.

Reply:
This is the same usage as that in standard `mlir-opt`, which doesn't have a specific header for test dialect: https://github.com/llvm/llvm-project/blob/0de1e3e787c6d78b87ccb865ba2f2daf8d8e4ecc/mlir/tools/mlir-opt/mlir-opt.cpp#L164

The actual header (and cpp) file is in the test files, not the source files: https://github.com/llvm/llvm-project/blob/0de1e3e787c6d78b87ccb865ba2f2daf8d8e4ecc/mlir/test/lib/Dialect/Test/TestDialect.h#L97, https://github.com/llvm/llvm-project/blob/0de1e3e787c6d78b87ccb865ba2f2daf8d8e4ecc/mlir/test/lib/Dialect/Test/TestDialect.cpp#L291